File: manuplationDigit/largestNumber.py
number = int(input("enter your 4 digit number :"))

lastDigit = number % 10
ThirdDigit = number // 10 % 10
SecondDigit = number // 100 % 10
firstDigit = number // 1000

if firstDigit >= SecondDigit and firstDigit >= ThirdDigit and firstDigit >= lastDigit:
    print(f"first digit is largest number {firstDigit}")

elif SecondDigit >= firstDigit and SecondDigit >= ThirdDigit and SecondDigit >= lastDigit:
    print(f"second digit is largest {SecondDigit}")

elif ThirdDigit >= firstDigit and ThirdDigit >= SecondDigit and ThirdDigit >= lastDigit:
    print(f"third digit is largest {ThirdDigit}")

else:
    print(f"fourth digit is largest {lastDigit}")

File: largest number.cpp
#include <iostream>

int main() {
    int num;
    std::cout << "Enter a 4-digit number: ";
    std::cin >> num;

    int d1 = num / 1000;
    int d2 = (num / 100) % 10;
    int d3 = (num / 10) % 10;
    int d4 = num % 10;

    int max = d1;
    if (d2 > max) max = d2;
    if (d3 > max) max = d3;
    if (d4 > max) max = d4;

    std::cout << "Largest digit: " << max << std::endl;
    return 0;
}